drivers/mtd/w25qxxxjv.c

```diff
@@ -628,14 +626,9 @@ static void w25qxxxjv_write_disable(FAR struct w25qxxxjv_dev_s *priv)
 
 static void w25qxxxjv_set_die(FAR struct w25qxxxjv_dev_s *priv, uint8_t die)
 {
-  char buff[1] =
-    {
-      die,
-    };
-
   w25qxxxjv_write_enable(priv);
   w25qxxxjv_command_write(priv->qspi, W25QXXXJV_SW_DIE_SELECT,
-                          (FAR const void *)buff, 1);
+                          (FAR const void *)&die, 1);
```

Comment:
I made this change not because of C89. It was C89 compatible before. It is because I started review the PR after it was merged, so instead of posting comments to merge PR I implemented my suggestions here.

Reply:
The only diff may be the alignment

Reply:
It seems it is this assertion https://github.com/apache/nuttx/blob/master/arch/arm/src/samv7/sam_qspi.c#L1401 that is raised when `&die` is used. So some alignment problem will be there.

Reply:
Yes. That is true, but `buf[1]` on stack will (most probably) give you proper alignment implicitly and that could give you an unexpected behavior in future.
I will be able to take a look here on Monday and provide a proper fix (that will be to use a union for `uint32_t` and 4 bytes `uint8_t` array, so it will ensure that address will be properly aligned, or explicit usage of align_data attribute).

Reply:
@michallenc please check if https://github.com/apache/nuttx/pull/9357 fix the issue and driver operates as expected